File: python/gpt_integration.py
import os
from typing import Optional

from dotenv import load_dotenv
from openai import OpenAI

load_dotenv()


class GPTIntegration:
    """Small stateful wrapper around the OpenAI Responses API."""

    def __init__(self, model: Optional[str] = None, client: Optional[OpenAI] = None):
        api_key = os.getenv("OPENAI_API_KEY")
        if client is None and not api_key:
            raise RuntimeError("OPENAI_API_KEY is required")

        self.client = client or OpenAI(api_key=api_key)
        self.model = model or os.getenv("GPT_MODEL", "gpt-5.5")
        self.previous_response_id: Optional[str] = None

    def chat(self, user_message: str, system_prompt: Optional[str] = None) -> str:
        if not isinstance(user_message, str) or not user_message.strip():
            raise ValueError("user_message must be a non-empty string")
        if system_prompt is not None and not isinstance(system_prompt, str):
            raise ValueError("system_prompt must be a string or None")

        request = {
            "model": self.model,
            "input": user_message.strip(),
            "store": True,
        }
        if system_prompt:
            request["instructions"] = system_prompt
        if self.previous_response_id:
            request["previous_response_id"] = self.previous_response_id

        response = self.client.responses.create(**request)
        self.previous_response_id = response.id
        text = (response.output_text or "").strip()
        if not text:
            raise RuntimeError("OpenAI returned an empty response")
        return text

    def reset_conversation(self) -> None:
        self.previous_response_id = None

    def analyze_text(self, text: str) -> str:
        if not isinstance(text, str) or not text.strip():
            raise ValueError("text must be a non-empty string")
        return self.chat(f"Please analyze the following text:\n\n{text.strip()}")

    def generate_content(self, topic: str, style: Optional[str] = None) -> str:
        if not isinstance(topic, str) or not topic.strip():
            raise ValueError("topic must be a non-empty string")
        prompt = f"Generate content about: {topic.strip()}"
        if style:
            prompt += f" in a {style.strip()} style."
        return self.chat(prompt)


if __name__ == "__main__":
    gpt = GPTIntegration()
    print(gpt.chat("Hello! How are you?"))

File: javascript/gpt-integration.js
import OpenAI from 'openai';
import dotenv from 'dotenv';
import path from 'node:path';
import { fileURLToPath } from 'node:url';

const __filename = fileURLToPath(import.meta.url);
const __dirname = path.dirname(__filename);
dotenv.config({ path: path.resolve(__dirname, '..', '.env') });

const client = new OpenAI({ apiKey: process.env.OPENAI_API_KEY });

export class GPTIntegration {
  constructor(model = process.env.GPT_MODEL || 'gpt-5.6-luna') {
    if (!process.env.OPENAI_API_KEY) {
      throw new Error('OPENAI_API_KEY is not configured');
    }
    this.model = model;
    this.conversationHistory = [];
  }

  async chat(userMessage, systemPrompt = null) {
    if (typeof userMessage !== 'string' || !userMessage.trim()) {
      throw new TypeError('userMessage must be a non-empty string');
    }

    const input = [];
    if (systemPrompt) {
      input.push({ role: 'system', content: systemPrompt });
    }
    input.push(...this.conversationHistory);
    input.push({ role: 'user', content: userMessage });

    const response = await client.responses.create({
      model: this.model,
      input,
    });

    const assistantMessage = response.output_text?.trim();
    if (!assistantMessage) {
      throw new Error('OpenAI returned an empty response');
    }

    this.conversationHistory.push({ role: 'user', content: userMessage });
    this.conversationHistory.push({ role: 'assistant', content: assistantMessage });

    return assistantMessage;
  }

  resetConversation() {
    this.conversationHistory = [];
  }

  async analyzeText(text) {
    if (typeof text !== 'string' || !text.trim()) {
      throw new TypeError('text must be a non-empty string');
    }
    return this.chat(`Please analyze the following text:\n\n${text}`);
  }

  async generateContent(topic, style = null) {
    if (typeof topic !== 'string' || !topic.trim()) {
      throw new TypeError('topic must be a non-empty string');
    }
    const prompt = style
      ? `Generate content about: ${topic} in a ${style} style.`
      : `Generate content about: ${topic}`;
    return this.chat(prompt);
  }
}

// Run the examples only when this file is executed directly.
if (process.argv[1] && path.resolve(process.argv[1]) === __filename) {
  const gpt = new GPTIntegration();
  const response = await gpt.chat('Hello! How are you?');
  console.log(`Assistant: ${response}`);
}
